packages/shylu/shylu_node/basker/src/shylubasker_matrix_def.hpp

```diff
@@ -528,52 +555,90 @@ namespace BaskerNS
       }
     }
     //init_value(val, nnz, (Entry) 0);
-    for(Int i = 0; i < nnz; ++i)
+    /*for(Int i = 0; i < nnz; ++i)
     {
       val(i) = 0;
-    }
+    }*/
+    //if (kid == 3)
+    //printf( " kid=%d: convert2D(%dx%d, nnz = %d, alloc = %d, scol = %d, ncol = %d)\n",kid,M.nrow,M.ncol,M.nnz,alloc, scol,ncol );
 
-    Int temp_count = 0;
+    const Entry zero(0.0);
 
+    anorm = abs(zero);
+    Int temp_count = 0;
     for(Int k = scol; k < scol+ncol; ++k)
     {
-      //note col_ptr[k-scol] contains the starting index
+      //note col_ptr[k-scol] contains the starting index (by find_2D_convert)
+      // i.e., start of the diagonal block for U, or start of the first off-diagonal block for L
       if(col_ptr(k-scol) == BASKER_MAX_IDX)
       {
         col_ptr(k-scol) = temp_count;
-        //printf("continue called, k: %d  \n", k);
+        /*if (kid == 1) {
```

Comment:
Remaining commented blocks can be removed in a later followup, similar for a few blocks of code below